modules/shortcodes/tweet.php

```diff
@@ -80,6 +80,17 @@ public static function jetpack_tweet_shortcode( $atts ) {
 			}
 		}
 
+		if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
```

Comment:
I think it may make sense to move this until after the WordPress.com conditional, right before the script that's hooked to `wp_footer` (since we don't want that in an AMP view).

I think ideally we'd want to have something like this:

```php
/** This action is documented in modules/widgets/social-media-icons.php */
do_action( 'jetpack_bump_stats_extras', 'embeds', 'tweet' );

if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
	...
	$output = sprintf(
		...
	);
} else {
	// Add Twitter widgets.js script to the footer.
	add_action( 'wp_footer', array( 'Jetpack_Tweet', 'jetpack_tweet_shortcode_script' ) );
}

return $output;
```

What do you think?

Reply:
That's a good idea, thanks for the detailed snippet.

How does https://github.com/Automattic/jetpack/pull/14054/commits/37c1df766dd4209e2ca86408658ba87aff97dff9 look?